src/Compilers/CSharp/Portable/Errors/LazyObsoleteDiagnosticInfo.cs

```diff
@@ -54,5 +55,31 @@ internal override DiagnosticInfo GetResolvedInfo()
 
             return _lazyActualObsoleteDiagnostic;
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is LazyObsoleteDiagnosticInfo lod)
+            {
+                return this.GetResolvedInfo().Equals(lod.GetResolvedInfo()) &&
```

Comment:
Do we need to compare `GetResolvedInfo()`? If all the fields are equal, won't `GetResolveInfo()` be equal as well?